engine/common/api_helpers/custom_rate_scoped_throttler.py

```diff
@@ -1,55 +1,30 @@
-from rest_framework.throttling import SimpleRateThrottle
+from django.conf import settings
+from ratelimit.utils import _split_rate
+from rest_framework.throttling import UserRateThrottle
 
 
-class CustomRateScopedThrottler(SimpleRateThrottle):
-    """
-    Abstract class to create throttlers with custom amount of seconds and custom scope.
-    The unique cache key will be generated by concatenating the
-    user id of the request, and the scope from get_scope() method.
+class CustomRateUserThrottler(UserRateThrottle):
+    """ """
 
-    Should not be used directly.
-    """
+    def parse_rate(self, rate):
+        "Use django ratelimit format to parse rate, i.e. '30/1m', instead of '30/m'"
+        return _split_rate(rate)
 
-    def __init__(self):
-        self.scope = self.get_scope()
-        self.num_requests, self.duration = self.get_throttle_limits()
-
-    def get_throttle_limits(self):
-        """
-        :return tuple requests/seconds
-        """
-        raise NotImplementedError
+    def allow_request(self, request, view):
+        # Override default rate limit, if organization id is specified in CUSTOM_RATELIMITS
+        custom_ratelimits = settings.CUSTOM_RATELIMITS
+        organization_id = str(request.user.organization_id)
+        if organization_id in custom_ratelimits:
+            self.rate = custom_ratelimits[organization_id]["public_api"]
+        self.num_requests, self.duration = self.parse_rate(self.rate)
 
-    def get_scope(self):
-        """
-        :return ratelimit scope
-        """
-        raise NotImplementedError
+        return super().allow_request(request, view)
 
-    def allow_request(self, request, view):
-        """
-        Overriden allow_request method.
-        The difference is that overriden method doesn't check rate property.
-        """
-
-        self.key = self.get_cache_key(request, view)
-        if self.key is None:
-            return True
-
-        self.history = self.cache.get(self.key, [])
-        self.now = self.timer()
-
-        # Drop any requests from the history which have now passed the
-        # throttle duration
-        while self.history and self.history[-1] <= self.now - self.duration:
-            self.history.pop()
-        if len(self.history) >= self.num_requests:
-            return self.throttle_failure()
-        return self.throttle_success()
 
+class CustomRateOrganizationThrottler(CustomRateUserThrottler):
```

Comment:
Should we set scope=org here to avoid clash with CustomRateUserThrottler.scope? Alternative idea is to introduce third class, containing allow_request and parse_rate logic and inherit CustomRateUserThrottler & CustomRateOrgThrottler with different scopes.